### ISSUE TITLE
aquamapsdata - Curated Data From AquaMaps.Org

### ISSUE THREAD

**Issue by mskyttner** [ropensci/software-review#421]
Submitting Author Name:  Markus Skyttner 
Submitting Author Github Handle: @mskyttner
Repository: https://github.com/raquamaps/aquamapsdata
Version submitted: 0.1.6
Editor:  @maurolepore 
Reviewers:  TBD 
 
Archive: TBD  
Version accepted: TBD   

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: aquamapsdata
Type: Package
Title: Curated Data From AquaMaps.Org
Date: 2020-11-21
Version: 0.1.6
Authors@R: c(
  person("Markus", "Skyttner", email = "markus.skyttner@nrm.se", role = c("aut", "cre")),
  person("Christina", "Garilao", role = c("ctb")))
Maintainer: Markus Skyttner <markus.skyttner@nrm.se>
Description: This package provides access to curated data from aquamaps.org published in 2020.
    It does not embed the data statically into the package, instead it attempts to download an SQLite database from a remote location.
    It has a function called download_db() which retrieves the data published by aquamaps.org as an SQLite database.
    It also has functions for accessing the data. The data is expected to be updated on the remote location approximately once per year.
License: file LICENSE
URL: https://github.com/raquamaps/aquamapsdata
BugReports: https://github.com/raquamaps/aquamapsdata/issues
Encoding: UTF-8
LazyData: true
RoxygenNote: 7.1.1
Imports:
    dplyr,
    dbplyr,
    RSQLite,
    R.utils,
    magrittr,
    stringi,
    purrr,
    DBI,
    RMySQL,
    duckdb,
    leaflet,
    rappdirs,
    raster,
    stringr,
    rlang,
    readr,
    rcrypt,
    sp,
    curl
Suggests: 
    knitr,
    httr,
    readxl,
    rmarkdown,
    xml2,
    DT,
    tidyr,
    testthat,
    covr
Depends:
    R (>= 3.0)
VignetteBuilder: knitr
Roxygen: list(markdown = TRUE)
SystemRequirements:
    rgdal

```

## Scope 

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) this package falls under: (Please check an appropriate box below. If you are unsure, we suggest you make a pre-submission inquiry.):

	- [x] data retrieval
	- [ ] data extraction
	- [ ] data munging
	- [ ] data deposition
	- [ ] workflow automation
	- [ ] version control
	- [ ] citation management and bibliometrics
	- [ ] scientific software wrappers
	- [ ] field and lab reproducibility tools
	- [ ] database software bindings
	- [x] geospatial data
	- [ ] text analysis
	
- Explain how and why the package falls under these categories (briefly, 1-2 sentences):

AquaMaps generates standardized computer-generated and fairly reliable large scale predictions of marine and freshwater species. The package provides functions for primarily accessing (and secondarily managing) curated data from aquamaps.org. It does not embed the data statically into the package itself; instead a function allows for a download of a compressed archive from archive.org, containing the full dataset in a single around 10+ GB large SQLite database file (close to 2 GB when compressed).

-   Who is the target audience and what are scientific applications of this package?  

The primary target audience is researchers studying ecology for aquatic species such as fish populations, when doing studies in ecology (federal agencies, academic researchers, industry users). Another audience consists of the people at aquamaps.org that manage the data. They have needs for workflows that allow them to expose the data using an R package and therefore need some functions to assist with converting and syncing data, making an extract (by minifying/slicing the data) and in general getting it in shape for being used in the package (including uploading data to archive.org, with approximately yearly frequency).

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

I haven't found any, but in the last couple of years I have looked mostly in a narrow fashion with regards to packages that provide data specifically from aquamaps.org. If widening the scope, the rfishbase and taxizedb packages are partly overlapping since these packages too provide for example for retrieving taxonomic information about aquatic species, but they're have a different scope or focus, since aquamapsdata is focused on providing spatial raster data for native habitats as predicted by the aquamaps algorithm.

-   (If applicable) Does your package comply with our [guidance around _Ethics, Data Privacy and Human Subjects Research_](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research)?

N/A

-   If you made a pre-submission enquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

## Technical checks

Confirm each of the following by checking the box.  

- [x] I have read the [guide for authors](https://devguide.ropensci.org/guide-for-authors.html) and [rOpenSci packaging guide](https://devguide.ropensci.org/building.html).

This package:

- [x] does not violate the Terms of Service of any service it interacts with. 
- [x] has a CRAN and OSI accepted license.
- [x] contains a [README with instructions for installing the development version](https://ropensci.github.io/dev_guide/building.html#readme). 
- [x] includes [documentation with examples for all functions, created with roxygen2](https://ropensci.github.io/dev_guide/building.html#documentation).
- [x] contains a vignette with examples of its essential functions and uses.
- [x] has a [test suite](https://ropensci.github.io/dev_guide/building.html#testing).
- [x] has [continuous integration](https://ropensci.github.io/dev_guide/ci.html), including reporting of test coverage using services such as Travis CI, Coveralls and/or CodeCov.

## Publication options

- [x] Do you intend for this package to go on CRAN?  
- [ ] Do you intend for this package to go on Bioconductor?  

- [ ] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:  

<details>
<summary>MEE Options</summary>  

- [ ] The package is novel and will be of interest to the broad readership of the journal. 
- [ ] The manuscript describing the package is no longer than 3000 words.
- [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
- (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
- (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
- (*Please do not submit your package separately to Methods in Ecology and Evolution*)

</details>

## Code of conduct

- [x] I agree to abide by [rOpenSci's Code of Conduct](https://ropensci.github.io/dev_guide/policies.html#code-of-conduct) during the review process and in maintaining my package should it be accepted.

**Comment by melvidoni** [comment]
Hello @mskyttner, thanks for the submission. Before proceeding, the Editors have some questions for you.

1.  We are not sure whether this package helps access data or distributes data. There is the impression the authors pre-prepared some dataset from the website they mention.

2. Are there no other more dynamic ways to get the data? Especially as in the future their prepared dataset might get stale.

3. Also, some editors had troubles accessing aquamaps.org, as they got a security alert. Can you double-check that the provided link is correct?

**Comment by mskyttner** [comment]
Hello @melvidoni,

1. Data for curated aquamaps as exposed by https://www.aquamaps.org is stored in a mysql database that powers that website, some more background about the data can be found [here](https://www.gbif.org/tool/81356/aquamaps-predicted-range-maps-for-aquatic-species). The aquamapsdata R package is a little bit like taxizedb in the sense that it takes almost all data in this database and makes it accessible from R to researchers, but it does not access the mysql database directly, instead it uses a compressed archive variant in SQLite format that gets downloaded from archive.org. So yes, it does help with accessing the data. But there is also a set of functions in the package to support distribution ("data management") that can be used by aquamaps staff themselves to generate this archive, ie for converting the whole mysql database into a SQLite database (with indexes and full text search capabilities) and to push a compressed archive to archive.org. So yes, it helps with distributing the data, too. These functions for "data management/distribution" is what is pulling down the test coverage stats. To summarize: yes, there are functions in the package for these different audiences - for data access (researchers, consumers of the data) and for distributing the data (aquamaps staff and their workflows as producers of the data). Can you clarify "There is the impression the authors pre-prepared some dataset from the website they mention.", which authors and website are you referring to? Some more info on the data scope is available here: https://raquamaps.github.io/aquamapsdata/articles/intro.html#scope
2. One could web scrape the aquamaps.org website. As in the archived https://github.com/ropensci-archive/zissou R package. But my understanding is that the aquamaps.org staff doesn't encourage that. There is no API yet, as far as I know.
3. Thanks for bringing this up, I checked https://www.ssllabs.com/ssltest/analyze.html?d=www.aquamaps.org and that seems to be OK, so it looks like I could change those links to https://www.aquamaps.org, but before doing that I'll first check with the aquamaps.org staff if they can also add aquamaps.org as a valid SNI name (it seems to resolve with a warning if skipping the www part, with the message "This server could not prove that it is aquamaps.org; its security certificate is from www.fishbase.de"; possibly this is a configuration that can be adjusted).

**Comment by melvidoni** [comment]
Thank you for the detailed response @mskyttner. Could you please clarify who updates the "compressed archive variant in SQLite format that gets downloaded from archive.org" and how often it is done?

**Comment by mskyttner** [comment]
Yes @melvidoni, this is done by AquaMaps staff using the "data management" functions in the package and scripts like those in [data-raw](https://github.com/raquamaps/aquamapsdata/blob/master/data-raw/am_upload_ia.R), a workflow which they have requested I developed support for in the package, in order to be able to do those updates more frequently, when data is changed or appended at the source. Historically the update frequency has been sporadic, with yearly updates. I have developed the package on basis of input from the AquaMaps team and will ask the staff there what their planned update frequency is and report back here going forward. Also, the database structure in the export / data distribution is "cleaned up" with regards to field names, scope/filters in order to be able to distribute the relevant data using a new more easily automatable workflow.

**Comment by melvidoni** [comment]
Thanks for the clarifications. @maurolepore will be the Handling Editor. He will be performing the editors' checks soon.

**Comment by maurolepore** [comment]
@mskyttner thanks for your submission! It's my pleasure to be the handling editor. 

Here are my editor checks ([guidelines](https://devguide.ropensci.org/editorguide.html#editorchecklist)). I'll start exploring potential reviewers but I would like to assign them only after you meet or justify against my requests (see section "Requests" below). This is to ensure the reviewers focus in the most interesting aspects of your work, and ultimately to help you maximize the quality of your package. Let me know when you are ready.

I also added a section "Observations". This is to support my requests and to guide you in finding a solution or argument against my requests.

Please let me know if you have any question or concern.

## Key editor checks

-   [ ] **Documentation**: The package has sufficient documentation available online (README, pkgdown docs) to allow for an assessment of functionality and scope without installing the package.
-   [ ] **Fit**: The package meets criteria for [fit](https://devguide.ropensci.org/policies.html#package-categories) and [overlap](https://devguide.ropensci.org/policies.html#overlap)
-   [ ] **Automated tests:** Package has a testing suite and is tested via Travis-CI or another CI service.
-   [ ] **License:** The package has a CRAN or OSI accepted license
-   [ ] **Repository:** The repository link resolves correctly

------------------------------------------------------------------------

## Editor comments

### Requests

FIT AND OVERLAP

-   [ ] If still relevant, please also argue explicitly how this package   fits in the category “geospatial data”.

-   [ ] Please refer to the  [guidelines](https://devguide.ropensci.org/policies.html#overlap)
    and discuss the potential overlap with other packages.

INSTALLATION

-   [ ] Do users need the system dependency `gpg`? If so, please add it   to DESCRIPTION.

INTEGRITY

-   [ ] Does DESCRIPTION need “rgdal” to include under Imports or   Suggests?

-   [ ] Please address (or justify against) the recommendations of  `goodpractice::gp()`.

METADATA

-   [ ] Please fix problematic URLs in README and DESCRIPTION.

-   [ ] Please review possibly misspelled words.

TESTS

-   [ ] Please thrive to increase test coverage or use [exclusion comments](https://github.com/r-lib/covr#exclusion-comments) if   appropriate.

CI

-   [ ] Please expand the configuration for continuous integration to   check the package with as many platforms and versions of R as   possible. DESCRIPTION claims to depend on R &gt;= 3.0 so it would be  nice to confirm this claim with automated checks.

STYLE

-   [ ] Please ensure source code is styled in a way that facilitates   reviews, e.g. via `styler::style_pkg()` and   `usethis::use_tidy_description()`.

### Observations

FIT AND OVERLAP

-   Here I see an argument for how the package fits in “data retrieval”   but not in “geospatial data”.

-   In README I see “A similar package which also provides the aquamaps   algorithm is available at <https://github.com/raquamaps/raquamaps>”;   but overlap is not discussed here.

INSTALLATION

-   Local install on my personal computer and personal library succeeds.

-   Local install on a container from rocker/verse fails with:

``` r
install(dependencies = TRUE, build_vignettes = TRUE)

Error: package or namespace load failed for ‘rcrypt’:
 .onLoad failed in loadNamespace() for 'rcrypt', details:
  call: fun(libname, pkgname)
  error: Please install GPG first or check if GPG can be run from the command line.
      The rcrypt package is just an interface to GPG. See https://gnupg.org/ for
      installation guidelines.
```

-   Installation from GitHub on my personal computer succeeds; but on a container from rocker/verse fails as above.

INTEGRITY

-   `devtools::check()` failed with a message suggesting a vignette   needs it, and it seems to not install with the package.

-   Also I experienced problems with codetools, but it might be a an odd  problem specific to my system.

``` r
Error in .doLoadActions(where, attach) : 
  error in load action .__A__.1 for package raster: loadModule(module = "spmod", what = TRUE, env = ns, loadNow = TRUE): Unable to load module "spmod": package ‘codetools’ was installed before R 4.0.0: please re-install it
Calls: <Anonymous> ... asNamespace -> loadNamespace -> <Anonymous> -> .doLoadActions
Execution halted
```

<details>
<summary>
Details
</summary>

``` r
devtools::check()
#> Updating aquamapsdata documentation
#> Loading aquamapsdata
#> 
#>   _.  _.      _. ._ _   _. ._   _  _|  _. _|_  _.
#>  (_| (_| |_| (_| | | | (_| |_) _> (_| (_|  |_ (_|
#>        |                   |
#> Writing NAMESPACE
#> Writing NAMESPACE
#> ── Building ──────────────────────────────────────────────────── aquamapsdata ──
#> Setting env vars:
#> ● CFLAGS    : -Wall -pedantic
#> ● CXXFLAGS  : -Wall -pedantic
#> ● CXX11FLAGS: -Wall -pedantic
#> ────────────────────────────────────────────────────────────────────────────────
#>      checking for file ‘/home/mauro/git/aquamapsdata/DESCRIPTION’ ...  ✓  checking for file ‘/home/mauro/git/aquamapsdata/DESCRIPTION’
#>   ─  preparing ‘aquamapsdata’:
#>      checking DESCRIPTION meta-information ...  ✓  checking DESCRIPTION meta-information
#>   ─  checking for LF line-endings in source and make files and shell scripts
#>   ─  checking for empty or unneeded directories
#>   ─  building ‘aquamapsdata_0.1.6.tar.gz’
#>      
#> ── Checking ──────────────────────────────────────────────────── aquamapsdata ──
#> Setting env vars:
#> ● _R_CHECK_CRAN_INCOMING_USE_ASPELL_: TRUE
#> ● _R_CHECK_CRAN_INCOMING_REMOTE_    : FALSE
#> ● _R_CHECK_CRAN_INCOMING_           : FALSE
#> ● _R_CHECK_FORCE_SUGGESTS_          : FALSE
#> ● NOT_CRAN                          : true
#> ── R CMD check ─────────────────────────────────────────────────────────────────
#> * using log directory ‘/tmp/Rtmp23ztZ6/aquamapsdata.Rcheck’
#> * using R version 4.0.3 (2020-10-10)
#> * using platform: x86_64-pc-linux-gnu (64-bit)
#> * using session charset: UTF-8
#> * using options ‘--no-manual --as-cran’
#> * checking for file ‘aquamapsdata/DESCRIPTION’ ... OK
#> * checking extension type ... Package
#> * this is package ‘aquamapsdata’ version ‘0.1.6’
#> * package encoding: UTF-8
#> * checking package namespace information ... OK
#> * checking package dependencies ... OK
#> * checking if this is a source package ... OK
#> * checking if there is a namespace ... OK
#> * checking for executable files ... OK
#> * checking for hidden files and directories ... OK
#> * checking for portable file names ... OK
#> * checking for sufficient/correct file permissions ... OK
#> * checking serialization versions ... OK
#> * checking whether package ‘aquamapsdata’ can be installed ... ERROR
#> Installation failed.
#> See ‘/tmp/Rtmp23ztZ6/aquamapsdata.Rcheck/00install.out’ for details.
#> * DONE
#> 
#> Status: 1 ERROR
#> See
#>   ‘/tmp/Rtmp23ztZ6/aquamapsdata.Rcheck/00check.log’
#> for details.
#> ── R CMD check results ───────────────────────────────── aquamapsdata 0.1.6 ────
#> Duration: 28.4s
#> 
#> > checking whether package ‘aquamapsdata’ can be installed ... ERROR
#>   See below...
#> 
#> ── Install failure ─────────────────────────────────────────────────────────────
#> 
#> * installing *source* package ‘aquamapsdata’ ...
#> ** using staged installation
#> ** R
#> ** data
#> *** moving datasets to lazyload DB
#> ** exec
#> ** inst
#> ** byte-compile and prepare package for lazy loading
#> Error in .doLoadActions(where, attach) : 
#>   error in load action .__A__.1 for package raster: loadModule(module = "spmod", what = TRUE, env = ns, loadNow = TRUE): Unable to load module "spmod": package ‘codetools’ was installed before R 4.0.0: please re-install it
#> Calls: <Anonymous> ... asNamespace -> loadNamespace -> <Anonymous> -> .doLoadActions
#> Execution halted
#> ERROR: lazy loading failed for package ‘aquamapsdata’
#> * removing ‘/tmp/Rtmp23ztZ6/aquamapsdata.Rcheck/aquamapsdata’
#> 
#> 1 error x | 0 warnings ✓ | 0 notes ✓
#> Error: R CMD check found ERRORs
```

</details>

-   All tests pass.

``` r
devtools::test()
#> Loading aquamapsdata
#> 
#>   _.  _.      _. ._ _   _. ._   _  _|  _. _|_  _.
#>  (_| (_| |_| (_| | | | (_| |_) _> (_| (_|  |_ (_|
#>        |                   |
#> Testing aquamapsdata
#> ✓ |  OK F W S | Context
#> ⠏ |   0       | csquare                                                         ⠏ |   0       | Csquare Conversions                                             ✓ |  12       | Csquare Conversions
#> ⠏ |   0       | name_search                                                     ⠏ |   0       | Name Search                                                     ⠧ |   8       | Name Search                                                     ✓ |   9       | Name Search [0.1 s]
#> 
#> ══ Results ═════════════════════════════════════════════════════════════════════
#> Duration: 0.2 s
#> 
#> [ FAIL 0 | WARN 0 | SKIP 0 | PASS 21 ]
```

-   I see some good practices are not observed:

<details>
<summary>
Details
</summary>

``` r
goodpractice::gp()
#> Preparing: covr
#> Preparing: cyclocomp
#> Warning in MYPREPS[[prep]](state, quiet = quiet): Prep step for cyclomatic
#> complexity failed.
#> Preparing: description
#> Preparing: lintr
#> Preparing: namespace
#> Preparing: rcmdcheck
#> Warning in MYPREPS[[prep]](state, quiet = quiet): Prep step for rcmdcheck
#> failed.
#> ── GP aquamapsdata ─────────────────────────────────────────────────────────────
#> 
#> It is good practice to
#> 
#>   ✖ write unit tests for all functions, and all package code in
#>     general. 25% of code lines are covered by test cases.
#> 
#>     R/aquamapsdata-package.R:89:NA
#>     R/aquamapsdata-package.R:90:NA
#>     R/aquamapsdata-package.R:91:NA
#>     R/aquamapsdata-package.R:92:NA
#>     R/aquamapsdata-package.R:93:NA
#>     ... and 427 more lines
#> 
#>   ✖ omit "Date" in DESCRIPTION. It is not required and it gets invalid
#>     quite often. A build date will be added to the package when you
#>     perform `R CMD build` on it.
#>   ✖ avoid long code lines, it is bad for readability. Also, many people
#>     prefer editor windows that are about 80 characters wide. Try make
#>     your lines shorter than 80 characters
#> 
#>     R/spatial.R:57:1
#>     R/spatial.R:70:1
#>     R/spatial.R:83:1
#> 
#>   ✖ avoid sapply(), it is not type safe. It might return a vector, or a
#>     list, depending on the input data. Consider using vapply() instead.
#> 
#>     R/csquare.R:129:11
#>     R/csquare.R:130:11
#>     R/csquare.R:131:11
#> 
#>   ✖ not import packages as a whole, as this can cause name clashes
#>     between the imported packages. Instead, import only the specific
#>     functions you need.
#> ────────────────────────────────────────────────────────────────────────────────
```

</details>

METADATA

-   I see some problematic URLs

``` r
> urlchecker::url_check()
! Warning: README.md:84:19 Moved
Unported License](http://creativecommons.org/licenses/by-nc/3.0/):
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                  https://creativecommons.org/licenses/by-nc/3.0/
x Error: README.md:32:2 Error: SSL: no alternative certificate subject name matches target host name 'aquamaps.org'
<https://aquamaps.org>, but the `aquamapsdata` package offers an a
 ^~~~~~~~~~~~~~~~~~~~
x Error: README.md:71:43 Error: SSL: no alternative certificate subject name matches target host name 'aquamaps.org'
`aquamapsdata` provides data output from <https://aquamaps.org>; when
                                          ^~~~~~~~~~~~~~~~~~~~
x Error: man/am_meta.Rd:18:7 Error: SSL: no alternative certificate subject name matches target host name 'aquamaps.org'
\href{https://aquamaps.org/}{AquaMaps}
      ^~~~~~~~~~~~~~~~~~~~~
```

SPELLCHECK

* I see unknown or possibly misspelled words:

<details>
<summary>
Details
</summary>

``` r
spelling::spell_check_package()
#> DESCRIPTION does not contain 'Language' field. Defaulting to 'en-US'.
#>   WORD                    FOUND IN
#> AGPL                    README.md:129
#>                         README.Rmd:103
#> alf                     intro.Rmd:154
#> amd                     NEWS.md:46
#> aquamaps                aquamapsdata-package.Rd:8
#>                         con_am.Rd:18
#>                         db_minify.Rd:5
#>                         description:1,3
#>                         README.md:69,84
#>                         README.Rmd:75,81
#>                         intro.Rmd:146
#> AquaMaps                am_citation.Rd:16,18,23,24
#>                         am_meta.Rd:18,28,41,47,52
#>                         am_search_exact.Rd:36
#>                         db_minify.Rd:17
#>                         title:1
#>                         README.md:28,81,88,93,100,114
#>                         README.Rmd:41,81,85,88,92,98
#>                         intro.Rmd:12,231,233,235,237,239
#> AquaMaps’             intro.Rmd:231
#> aquamapsdata’         README.md:130
#> aquamapsdb              am_use_offline_db.Rd:12
#>                         aquamapsdata-package.Rd:12
#> au                      csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#> Barile                  README.md:81
#>                         README.Rmd:81
#> bathydemersal           am_search_exact.Rd:66
#> bathypelagic            am_search_exact.Rd:66
#> biologie                am_citation.Rd:32
#>                         README.md:107
#>                         README.Rmd:94
#> Bluespotted             intro.Rmd:98
#> Caranx                  intro.Rmd:131
#> Catalogue               am_search_exact.Rd:37
#> catalogueoflife         am_search_exact.Rd:37
#> cells’                intro.Rmd:231
#> chunkwise               intro.Rmd:294
#> cmar                    csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#> CMD                     NEWS.md:10
#>                         README.md:5
#>                         README.Rmd:9
#> Codecov                 README.md:10
#>                         README.Rmd:11
#> csiro                   csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#> csq                     csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#> csquare                 csquare_from_dd.Rd:5,21
#>                         csquare_to_dd.Rd:5,10,16
#> csquarecode             NEWS.md:22
#> CsquareCode             am_csc_from_extent.Rd:19,22
#>                         intro.Rmd:178,208,210
#> CsquareCodes            am_species_in_csc.Rd:10,16
#>                         am_species_per_csc.Rd:10
#> csquares                csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#> csv                     NEWS.md:42
#> datadir                 intro.Rmd:259
#> de                      am_citation.Rd:31,32
#>                         README.md:107
#>                         README.Rmd:94
#> degress                 csquare_from_dd.Rd:14
#> DepthPrefMin            am_search_exact.Rd:71
#> dplyr                   aquamapsdata-package.Rd:14
#> duckdb                  con_am.Rd:10
#>                         default_db.Rd:10
#> eval                    NEWS.md:16
#> extdata                 am_use_offline_db.Rd:5,12
#>                         con_am.Rd:10
#>                         default_db.Rd:10,17
#>                         NEWS.md:32
#> faq                     csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#> FB                      am_search_exact.Rd:94
#> FC                      am_map_leaflet.Rd:19
#> fcn                     am_custom_query.Rd:14
#> Fis                     am_search_exact.Rd:38
#> FishBase                am_search_exact.Rd:40,46,53,66,69,75,80,82,86,94
#> freiburg                am_citation.Rd:32
#>                         README.md:107
#>                         README.Rmd:94
#> Froese                  am_citation.Rd:31
#>                         README.md:81,107
#>                         README.Rmd:81,94
#>                         intro.Rmd:239
#> fts                     am_search_fuzzy.Rd:11
#> FTS                     NEWS.md:28
#>                         intro.Rmd:302
#> GameFish                am_search_exact.Rd:68
#> geomar                  am_citation.Rd:31
#>                         README.md:107
#>                         README.Rmd:94
#> github                  README.md:36,39
#>                         README.Rmd:45,47
#> gz                      NEWS.md:46
#> hcaf                    am_meta.Rd:38,50
#>                         NEWS.md:30
#>                         intro.Rmd:210,233
#> HCAF                    intro.Rmd:158
#> HSPEC                   am_meta.Rd:39
#> hspen                   am_meta.Rd:47
#>                         NEWS.md:30
#>                         intro.Rmd:190
#> HSPEN                   am_search_exact.Rd:72
#>                         intro.Rmd:192
#> htm                     csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#> http                    am_search_fuzzy.Rd:11
#>                         csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#> https                   aquamapsdata-package.Rd:12
#> ie                      am_species_per_csc.Rd:5
#> ile                     intro.Rmd:154
#> IPCC                    am_meta.Rd:54
#> IUCN                    am_meta.Rd:36
#>                         am_search_exact.Rd:88,90,92
#> Kaschner                am_citation.Rd:32
#>                         README.md:81,107
#>                         README.Rmd:81,94
#>                         intro.Rmd:239
#> Kesner                  README.md:81
#>                         README.Rmd:81
#> LME                     intro.Rmd:210
#> LOICZID                 which_cells_in_raster.Rd:5,19
#> loiczids                which_cells_in_raster.Rd:16
#> lon                     csquare_to_dd.Rd:13
#> md                      am_citation.Rd:10
#>                         intro.Rmd:120
#> minified                am_use_offline_db.Rd:12
#>                         db_minify_path.Rd:5,10
#>                         default_db.Rd:17
#>                         intro.Rmd:12
#> Minify                  db_minify.Rd:5
#> mysql                   con_am.Rd:10
#>                         db_sync.Rd:35
#>                         default_db.Rd:10
#> NonCommercial           README.md:90
#>                         README.Rmd:86
#> occ                     NEWS.md:30
#> occuring                am_species_in_csc.Rd:5
#>                         intro.Rmd:178
#> occurrencecells         am_meta.Rd:56
#> pkgdown                 NEWS.md:12
#> pre                     con_am.Rd:17
#>                         NEWS.md:44
#> programmatically        README.md:31
#>                         README.Rmd:43
#> Rainer                  am_citation.Rd:31
#>                         README.md:107
#>                         README.Rmd:94
#>                         intro.Rmd:239
#> RCP                     am_meta.Rd:54,55
#>                         intro.Rmd:235
#> Rees                    README.md:81
#>                         README.Rmd:81
#> rfroese                 am_citation.Rd:31
#>                         README.md:107
#>                         README.Rmd:94
#> Rius                    README.md:81
#>                         README.Rmd:81
#> Rmarkdown               intro.Rmd:120
#> ROpenSci                README.md:135
#>                         README.Rmd:106
#> RSQLite                 NEWS.md:16
#>                         README.md:63
#>                         README.Rmd:71
#> Salmo                   NEWS.md:16
#> SeaLifeBase             am_search_exact.Rd:40,46,53,66,75,83,86,94
#> Segschneider            README.md:81
#>                         README.Rmd:81
#> SLB                     am_search_exact.Rd:94
#> species’environmental intro.Rmd:190
#> SpeciesID               am_species_per_csc.Rd:15
#> speciesoccursum         am_meta.Rd:32
#> sql                     am_custom_query.Rd:14
#> sqlite                  am_custom_query.Rd:12
#>                         am_db_sqlite.Rd:5,13
#>                         am_search_fuzzy.Rd:11
#>                         con_am.Rd:10,11
#>                         db_minify_path.Rd:5,10
#>                         default_db.Rd:10,17
#>                         NEWS.md:42
#> stringi                 NEWS.md:24
#> stringr                 NEWS.md:24
#> tibble                  am_csc_from_extent.Rd:19
#>                         am_search_exact.Rd:97
#>                         am_search_fuzzy.Rd:14
#> tidyverse               README.md:135
#>                         README.Rmd:106
#>                         intro.Rmd:42
#> traumatogenic           am_search_exact.Rd:74
#> travis                  NEWS.md:38
#> trevally                intro.Rmd:98
#> trutta                  NEWS.md:16
#> Unported                README.md:90
#>                         README.Rmd:86
#> uthority                intro.Rmd:154
#> venonous                am_search_exact.Rd:75
#> WGS                     am_csc_from_extent.Rd:10,12,14,16
#>                         rasterize_coords.Rd:18
#>                         which_cells_in_raster.Rd:10
#> www                     am_search_exact.Rd:37
#>                         am_search_fuzzy.Rd:11
#>                         csquare_from_dd.Rd:22
#>                         csquare_to_dd.Rd:17
#>                         README.md:84
#>                         README.Rmd:81
#> xmax                    am_csc_from_extent.Rd:12
#> xmin                    am_csc_from_extent.Rd:10
#> ymax                    am_csc_from_extent.Rd:16
#> ymin                    am_csc_from_extent.Rd:14
#> yml                     NEWS.md:38
```

</details>

TESTS

In some files test coverage is far under 80%.

``` r
covr::package_coverage()
#> aquamapsdata Coverage: 25.90%
#> R/aquamapsdata-package.R: 0.00%
#> R/download.R: 0.00%
#> R/spatial.R: 0.00%
#> R/db.R: 5.17%
#> R/data.R: 30.10%
#> R/csquare.R: 97.92%
#> R/zzz.R: 100.00%
```

CI

-   The continuous integration workflow for GitHub Actions runs R CMD  check on only on ubuntu-20.04 and R-release.

-   The continuous integration configuration on Travis also seems to   check the package with only one platform.

STYLE

-   `styler::style_pkg()` changes most files and detects one syntax  error:

``` r
> styler::style_pkg()
Styling  20  files:
 R/aquamapsdata-package.R          ℹ 
 R/csquare.R                       ℹ 
 R/data.R                          ℹ 
 R/db.R                            ℹ 
 R/download.R                      ℹ 
 R/pipe.R                          ✓ 
 R/spatial.R                       ℹ 
 R/utils.R                         ℹ 
 R/zzz.R                           ℹ 
 tests/testthat.R                  ✓ 
 tests/testthat/test_csquare.R     ℹ 
 tests/testthat/test_name_search.R ℹ 
 data-raw/am_meta.R                ℹ 
 data-raw/am_minify_db.R           ℹ 
 data-raw/am_search_boilerplate.R  ℹ 
 data-raw/am_slice_db.R            ℹ 
 data-raw/am_upload_ia.R           ! 
 data-raw/convert_csvs_to_db.R     ℹ 
 data-raw/head_hundred.R           ℹ 
 data-raw/onboarding_dev.R         ℹ 
────────────────────────────────────────
Status  Count   Legend 
✓   2   File unchanged.
ℹ   17  File changed.
x   1   Styling threw an error.
────────────────────────────────────────
Please review the changes carefully!
Warning message:
When processing data-raw/am_upload_ia.R: <text>:1:8: unexpected '{'
1: dontrun{
           ^ 
```

------------------------------------------------------------------------

@mskyttner, could you please add the rOpenSci under review badge to your README?

--

Reviewers:   
Due date:

**Comment by mskyttner** [comment]
Thanks @maurolepore for the feedback! I'm breaking those requests/items up into steps/issues now and will ping back once ready.

**Comment by maurolepore** [comment]
Hi @mskyttner, just checking how you are going. What would be your estimated time to address the items I mentioned above? There is no rush. I only ask to see if in the meantime I can help with other submissions.

**Comment by mskyttner** [comment]
Hi @maurolepore thanks for checking up  on me, I have been off for a while due to some health issues with my elderly parents but getting back now. I have about half of the editor check issues raised adressed (I think) here https://github.com/raquamaps/aquamapsdata/pull/32 but probably need another week to work through the CI / test coverage / style. I tried to use a rocker/geospatial base image rather than rocker/tidyverse base in the CI setup due to some issues with rgdal installation etc, still struggling a bit with getting that sorted out for all the platforms and versions of R etc. Will try to look more in depth at links with examples how this can be set up ^^.

**Comment by maurolepore** [comment]
Thanks for letting me know. I'm sorry to hear that. I'll apply the "holding" label and try help elsewhere. Let me know when you are ready.

**Comment by maurolepore** [comment]
@mskyttner, just touching base. Are you okay staying on hold?

**Comment by maurolepore** [comment]
@mskyttner I really hope you and your family are well. I realize I touched base a bit too early. The first revision of the 'holding' label will formally be on May 22.

> The holding status will be revisited every 3 months, and after one year the issue will be closed.
> --https://devdevguide.netlify.app/policies.html#policiesreviewprocess

**Comment by maurolepore** [comment]
Hi @mskyttner, it's about the time when we should review the "holding" status. Shall we keep it or are you ready to resume work on this submission?

--

The author can choose to have their submission put on hold (editor applies the holding label). The holding status will be revisited every 3 months, and after one year the issue will be closed. --
-- https://devdevguide.netlify.app/policies.html#policiesreviewprocess

**Comment by emilyriederer** [comment]
Hi @mskyttner  ! We are doing a sweep of stale review issues. Since this review has been open and inactive for so long, much may have changed including author, editor, and reviewer bandwidth and ever-evolving rOpenSci best practices. As such, I'm closing this issue. If you still have interest and capacity, we would welcome you to open a new submission issue!
